operations/helm/charts/grafana-agent/templates/hpa.yaml

```diff
@@ -1,4 +1,16 @@
 {{- if and (or (eq .Values.controller.type "deployment") (eq .Values.controller.type "statefulset" ))  .Values.controller.autoscaling.enabled }}
+{{- if not (empty .targetMemoryUtilizationPercentage)}} 
+  {{- $name := .Values.agent.resources.requests | required ".Values.agent.resources.requests is required when using autoscaling." -}}
+  {{- $name := .Values.agent.resources.requests.memory | required ".Values.agent.resources.requests.memory is required when using autoscaling based on memory utilization." -}}
+  {{- $name := .Values.configReloader.resources.requests  | required ".Values.configReloader.resources.requests is required when using autoscaling." -}}
+  {{- $name := .Values.configReloader.resources.requests.memory  | required ".Values.configReloader.resources.requests.memory is required when using autoscaling based on memory utilization." -}}
+{{- end}}
+{{- if not (empty .targetCPUUtilizationPercentage)}} 
+  {{- $name := .Values.agent.resources.requests | required ".Values.agent.resources.requests is required when using autoscaling." -}}
+  {{- $name := .Values.agent.resources.requests.cpu | required ".Values.agent.resources.requests.cpu is required when using autoscaling based on cpu utilization." -}}
+  {{- $name := .Values.configReloader.resources.requests  | required ".Values.configReloader.resources.requests is required when using autoscaling." -}}
+  {{- $name := .Values.configReloader.resources.requests.cpu  | required ".Values.configReloader.resources.requests.cpu is required when using autoscaling based on cpu utilization." -}}
+{{- end}}
```

Comment:
This is a great idea, I just wish it didn't have to be so long 😢

Reply:
```go
  {{- .Values.agent.resources.requests | required ".Values.agent.resources.requests is required when using autoscaling." -}}
  {{- .Values.agent.resources.requests.cpu | required ".Values.agent.resources.requests.cpu is required when using autoscaling based on cpu utilization." -}}
  {{- .Values.configReloader.resources.requests | required ".Values.configReloader.resources.requests is required when using autoscaling." -}}
  {{- .Values.configReloader.resources.requests.cpu | required ".Values.configReloader.resources.requests.cpu is required when using autoscaling based on cpu utilization." -}}
```

would still be pretty long but a little bit shorter :)

Reply:
Good catch, I probably copied from an Helm docs example; we don't make use of it here. Removing it.

Reply:
Hmm, when I remove the variable, it reports the following error. Maybe the example had the local variable to avoid actually rendering the value it if it was present, I'll re-read the docs.

```
$ helm upgrade --install --namespace agent grafana-agent .
Error: UPGRADE FAILED: YAML parse error on grafana-agent/templates/hpa.yaml: error converting YAML to JSON: yaml: line 2: mapping values are not allowed in this context
```

Reply:
Never mind, I see that required returns the value, so you do actually need to bind it to a temporary variable.